src/Offer/ReadModel/Metadata/OfferMetadataProjector.php

```diff
@@ -0,0 +1,125 @@
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($eventCreated->getEventId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
+    }
+
+    public function applyPlaceCreated(PlaceCreated $placeCreated, DomainMessage $domainMessage)
+    {
+        try {
+            $offerMetadata = $this->repository->get($placeCreated->getPlaceId());
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($placeCreated->getPlaceId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
+    }
+
+    public function applyEventCopied(EventCopied $eventCopied, DomainMessage $domainMessage)
+    {
+        try {
+            $offerMetadata = $this->repository->get($eventCopied->getItemId());
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($eventCopied->getItemId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
+    }
+
+    public function applyEventImportedFromUDB2(EventImportedFromUDB2 $eventImportedFromUDB2, DomainMessage $domainMessage)
+    {
+        try {
+            $offerMetadata = $this->repository->get($eventImportedFromUDB2->getEventId());
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($eventImportedFromUDB2->getEventId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
+    }
+
+    public function applyPlaceImportedFromUDB2(PlaceImportedFromUDB2 $placeImportedFromUDB2, DomainMessage $domainMessage)
+    {
+        try {
+            $offerMetadata = $this->repository->get($placeImportedFromUDB2->getActorId());
+        } catch (EntityNotFoundException $e) {
+            $offerMetadata = OfferMetadata::default($placeImportedFromUDB2->getActorId());
+        }
+
+        $createdByApiConsumer = $this->getCreatedByApiConsumerFromMetadata($domainMessage->getMetadata());
+        $offerMetadata = $offerMetadata->withCreatedByApiConsumer($createdByApiConsumer);
+
+        $this->repository->save($offerMetadata);
+    }
+
+    private function getCreatedByApiConsumerFromMetadata(Metadata $metadata): string
```

Comment:
Maybe an `CreatedByApiConsumer` value object can help here with `other` and `unknown`?

Reply:
Yeah, I wasn't sure if this was enough to add another VO to the codebase as it's only used in this spot.